types/assert/ts3.6/index.d.ts

```diff
@@ -1,55 +1,77 @@
             expected?: any;
+            /** The `operator` property on the error instance. */
             operator?: string;
-            stackStartFunction?: () => void;
+            /** If provided, the generated stack trace omits frames before this function. */
+            stackStartFn?: Function;
         });
     }
 
+    type AssertPredicate = RegExp | (new () => object) | ((thrown: any) => boolean) | object | Error;
+
+    function fail(message?: string | Error): never;
+    /** @deprecated since v10.0.0 - use fail([message]) or other assert functions instead. */
+    function fail(
+        actual: any,
+        expected: any,
+        message?: string | Error,
+        operator?: string,
+        stackStartFn?: Function,
+    ): never;
+    function ok(value: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use strictEqual() instead. */
+    function equal(actual: any, expected: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use notStrictEqual() instead. */
+    function notEqual(actual: any, expected: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use deepStrictEqual() instead. */
+    function deepEqual(actual: any, expected: any, message?: string | Error): void;
+    /** @deprecated since v9.9.0 - use notDeepStrictEqual() instead. */
+    function notDeepEqual(actual: any, expected: any, message?: string | Error): void;
+    function strictEqual(actual: any, expected: any, message?: string | Error): void;
+    function notStrictEqual(actual: any, expected: any, message?: string | Error): void;
+    function deepStrictEqual(actual: any, expected: any, message?: string | Error): void;
+    function notDeepStrictEqual(actual: any, expected: any, message?: string | Error): void;
+
+    function throws(block: () => any, message?: string | Error): void;
+    function throws(block: () => any, error: AssertPredicate, message?: string | Error): void;
+    function doesNotThrow(block: () => any, message?: string | Error): void;
+    function doesNotThrow(block: () => any, error: RegExp | Function, message?: string | Error): void;
+
+    function ifError(value: any): void;
+
+    function rejects(block: (() => Promise<any>) | Promise<any>, message?: string | Error): Promise<void>;
+    function rejects(
+        block: (() => Promise<any>) | Promise<any>,
+        error: AssertPredicate,
+        message?: string | Error,
+    ): Promise<void>;
+    function doesNotReject(block: (() => Promise<any>) | Promise<any>, message?: string | Error): Promise<void>;
+    function doesNotReject(
+        block: (() => Promise<any>) | Promise<any>,
+        error: RegExp | Function,
+        message?: string | Error,
+    ): Promise<void>;
+
+    function match(value: string, regExp: RegExp, message?: string | Error): void;
+    function doesNotMatch(value: string, regExp: RegExp, message?: string | Error): void;
+
     const strict: typeof assert;
```

Comment:
```suggestion
    const strict: Omit<
        typeof assert,
        | 'strict'
        | 'deepEqual'
        | 'notDeepEqual'
        | 'equal'
        | 'notEqual'
    > & {
        (value: any, message?: string | Error): void;
        strict: typeof strict;
        deepEqual: typeof deepStrictEqual;
        notDeepEqual: typeof notDeepStrictEqual;
        equal: typeof strictEqual;
        notEqual: typeof notStrictEqual;
    };
```

Reply:
Good suggestion but `Omit<>` was added in 3.5 and these types are still for Minimum TypeScript Version < 3.5?